File: decorator/decorator.py
class Coffee:
    def cost(self):
        return 5
class SimpleCoffee(Coffee):
    pass

class CoffeeDecorator(Coffee):
    def __init__(self,coffee):
        self._coffee = coffee

    def cost(self):
        return self._coffee.cost()

class MilkDecorator(CoffeeDecorator):
    def cost(self):
        return super().cost() + 2


simple_coffee = SimpleCoffee()
milk_coffee = MilkDecorator(simple_coffee)
print(milk_coffee.cost())

File: decorator/decorator.cs
using System;

public interface Coffee{
    int cost();
}

public class SimpleCoffee : Coffee{
    public int cost(){
        return 5;
    }
}

public class CoffeeDecorator : Coffee{
    protected Coffee coffee;
    public CoffeeDecorator(Coffee coffee){
        this.coffee = coffee;
    }
    public virtual int cost(){
        return coffee.cost();
    }
}

public class MilkDecorator : CoffeeDecorator{
    public MilkDecorator(Coffee coffee) : base(coffee){}
    public override int cost(){
        return base.cost() + 2;
    }
}

public class Program{
    public static void Main(string[] args){
        Coffee sc = new SimpleCoffee();
        Coffee md = new MilkDecorator(sc);
        Console.WriteLine(md.cost());
    }
}